prusti-encoder/src/encoders/viper_tuple.rs

```diff
@@ -150,7 +155,7 @@ impl TaskEncoder for ViperTupleEncoder {
                 domain: Some(vcx.alloc(vir::DomainData {
                     name: domain_name,
                     typarams: vcx.alloc_slice(&typaram_names),
-                    axioms: vcx.alloc_slice(&[axiom]),
+                    axioms: if task_key == &0 {&[]} else {vcx.alloc_slice(&[axiom])},
```

Comment:
Please extract this condition into a variable like `is_unit_tuple`. (Also spaces inside the braces are missing.)

Reply:
Agreed. @tillarnold this is not great code, if we have a unit slice then calculating and allocating everything for `axiom` is unnecessary - it's better to put this condition earlier on (see my latest push which fixes this)